app/controllers/crons_controller.rb

```diff
@@ -27,4 +15,31 @@ def create
       render(plain: "Unable to run #{taskname}", status: 500)
     end
   end
+
+  private
+
+  def runnable?
+    request.local? && configured_job_names.include?(taskname) && !running_same_job?
+  end
+
+  def configured_job_names
+    YAML.load_file(CRON_DEFINITIONS)['cron'].map { |d| d['name'] }
+  end
+
+  def taskname
+    @taskname ||= request.headers['X-Aws-Sqsd-Taskname']
+  end
+
+  def running_same_job?
+    res = false
+    IO.popen('ps ax') do |io|
```

Comment:
This won't work beyond one worker. Though we've only ever had one so probably fine.